Sources/StreamChat/Repositories/PollsRepository.swift

```diff
@@ -0,0 +1,269 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class PollsRepository {
+    let database: DatabaseContainer
+    let apiClient: APIClient
+    
+    init(database: DatabaseContainer, apiClient: APIClient) {
+        self.database = database
+        self.apiClient = apiClient
+    }
+    
+    func createPoll(
+        name: String,
+        allowAnswers: Bool? = nil,
+        allowUserSuggestedOptions: Bool? = nil,
+        description: String? = nil,
+        enforceUniqueVote: Bool? = nil,
+        maxVotesAllowed: Int? = nil,
+        votingVisibility: String? = nil,
+        options: [PollOption]? = nil,
+        custom: [String: RawJSON]? = nil,
```

Comment:
nit: Since the repository/updater is internal, I find it more useful not to make these properties optional, so that when we use it in the controllers, we don't forget to reason about all the available properties